### DISCUSSION TITLE
Known working Android phones/tablets & Android versions?

### DISCUSSION THREAD

**Discussion by johnwigley** [cagnulein/qdomyos-zwift#383] [Q&A]
I'm having severe trouble finding an Android device that will simultaneously operate in Bluetooth client & peripheral mode for longer than 30s.

I have tried a significant number of Android devices and the best I can find are devices that will operate with the "Relaxed Bluetooth" option for approx 30s, and then the bike (DKN AM-5i) disconnects and the controller resets.

Not working list, already tried with "Relaxed Bluetooth" both on/off:

Xiaomi Redmi Note 8 Pro (Android 10)
Xiaomi Redmi Note 8 Pro (Android 9)
Xiaomi Redmi Note 4 Pro (Android 7)  - Works for 30s
Lenovo P2 (Android 9)
Samsung Galaxy S5 (Android 6)
Samsung Tab S2 (Android 7)  - Works for 30s
Samsung Tab S3 (Android 8)
Samsung Tab S3 (Android 9)
Samsung Tab T285 (Android 5)  - Works for 30s

Most of these devices connect to and successfully operate the bike if the "Virtual Device" option is disabled for longer than 30s, but then Qdomyos-Zwift doesn't create a virtual FTMS device meaning it can't be used with Zwift.

Please could people add to this issue with examples of known working device/android combinations so that I can try and find a working one, Thanks!

**Comment by cagnulein** [comment]
For all the Samsung did you try to put a phone name short? For example
“bike”?
It’s so strange that no one works
Anyway it’s a good idea this one to keep a non working list

Il giorno mer 23 giu 2021 alle 22:13 johnwigley ***@***.***>
ha scritto:

> I'm having severe trouble finding an Android device that will
> simultaneously operate in Bluetooth client & peripheral mode for longer
> than 30s.
>
> I have tried a significant number of Android devices and the best I can
> find are devices that will operate with the "Relaxed Bluetooth" option for
> approx 30s, and then the bike (DKN AM-5i) disconnects and the controller
> resets.
>
> Not working list, already tried with "Relaxed Bluetooth" both on/off:
>
> Xiaomi Redmi Note 8 Pro (Android 10)
> Xiaomi Redmi Note 8 Pro (Android 9)
> Xiaomi Redmi Note 4 Pro (Android 7) - Works for 30s
> Lenovo P2 (Android 9)
> Samsung Galaxy S5 (Android 6)
> Samsung Tab S2 (Android 7) - Works for 30s
> Samsung Tab S3 (Android 8)
> Samsung Tab S3 (Android 9)
> Samsung Tab T285 (Android 5) - Works for 30s
>
> Most of these devices connect to and successfully operate the bike if the
> "Virtual Device" option is disabled for longer than 30s, but then
> Qdomyos-Zwift doesn't create a virtual FTMS device meaning it can't be used
> with Zwift.
>
> Please could people add to this issue with examples of known working
> device/android combinations so that I can try and find a working one,
> Thanks!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cagnulein/qdomyos-zwift/issues/381>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALYWAS6G3Y5YIRZOQKYJLTUI55VANCNFSM47GQABPQ>
> .
>
-- 
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud

**Comment by johnwigley** [comment]
Just tried shortening device BT name to "bike" on all three 30s working devices, made no difference unfortunately.

It seems as though you've seen the 30s connection bug before, and have tried the "relaxed bluetooth" option to deal with them, this seems like the same bug++. On these devices it won't work at all without "relaxed bluetooth" but then enabling it causes it to only work for 30s.

What is it that "relaxed bluetooth" actually does, and is there any possibility of repeating it on a 30s basis to keep the connection open? Thanks

**Comment by johnwigley** [comment]
Just having Googled looking for other people having problems with BLE 30s disconnects, came across an interesting StackOverflow on a similar topic. Seems that the 30s is a core BLE timeout for incomplete transactions.

https://stackoverflow.com/questions/54898592/bluetoothgattserver-is-always-disconnecting-after-30-seconds

I had wondered if there was something else different about this DKN Am-5i Iconsole protocol from the usual protocol, but that wouldn't explain why it stays connected IF virtual device is disabled.

What's interesting though is that during the 30s that it stays connected then Qdomyos is able to successfully control the resistance and read sensor data coming from the bike, so that suggests that all needed protocol communication is correctly implemented for this bike controller.

Is it possible that something is going on like transaction event listeners are not being notified of outstanding transactions when the device is in concurrent central and peripheral mode, and regular polling rather than relying on event listeners might help in this situation of buggy BT stacks?

**Comment by cagnulein** [comment]
#366 I will check again this hci snoof. Maybe it wants something else

**Comment by cagnulein** [comment]
i read again your ticket, and i forgot about the fact that if you disable the virtual device everything works smoothly. So it's the problem about the central/peripheral at the same time issue.
In the facebook group you can find a lot of useful informations about the device that FOR SURE are working.

I'm moving this to a discussion, maybe it could be useful to add this in the wiki page

**Comment by lifof** [comment]
Hello there

From my experience the cause of the disconnections that occurs during ZWift sessions was due to connecting two peripherals to my phone ( good old Galaxy S7 ) via BLE, the first was an Elliptical and the second was an external BLE Heart-rate monitor.

So what I did was to disconnect the Heart-rate monitor from the qDomyos-zwift app then everything worked just fine, long sessions without interruptions, as for the Heart-rate monitor you can connect it to your ZWift app.

I hope this helps you in some way.
